pkg/lib/serverTime.js

```diff
@@ -321,13 +316,166 @@
                     console.warn("failed to load time servers", error);
                     return result;
                 });
-    };
+    }
+
+    function set_ntp_config_timesyncd(config) {
+        const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
+                                                    { superuser: "try" });
 
-    self.set_custom_ntp = function (servers, enabled) {
-        const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${enabled ? "" : "#"}NTP=${servers.join(" ")}\n`;
+        const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${config.enabled ? "" : "#"}NTP=${config.servers.join(" ")}\n`;
 
         return cockpit.spawn(["mkdir", "-p", "/etc/systemd/timesyncd.conf.d"], { superuser: "try" })
                 .then(() => custom_ntp_config_file.replace(text));
+    }
+
+    function get_ntp_config_chronyd() {
+        const chrony_file = cockpit.file("/etc/chrony.conf", { superuser: "try" });
+
+        return chrony_file.read()
+                .then(function(text) {
+                    const result = {
+                        backend: "chronyd",
+                        supported: true,
+                        standard: [],
+                        sources: [],
+                        extra: []
+                    };
+
+                    let in_cockpit_section = false;
+                    for (const line of text.split("\n")) {
+                        if (line == "# Added by Cockpit")
+                            in_cockpit_section = true;
+                        else {
+                            const fields = line.split(" ");
+                            if (fields[0] == "server") {
+                                if (in_cockpit_section)
+                                    result.extra.push(fields[1]);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17992-20230113-125341-c8c3ecb6-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)